### PR TITLE
Add diagnostic collection workflow task

### PR DESCRIPTION
**Pull request by danbudris**
*Issue #, if available:*

*Description of changes:*

Add a new workflow task, `CollectDiagnosticsTask`, which invokes the SaveLogs methods of the Cluster Manager.

Run this `CollectDiagnosticsTask` after an error and prior to exiting; additionally, invoke it in 'terminal' tasks such as `DeleteBootstrapClusterTask`, `DeleteKindClusterTask`, and `deleteManagementCluster`, prior to cleanup and exit, if the commandContext has an error set.

This allows us to determine when and where the diagnostic bundle is collected. This specifically solve the issue of invoke the diagnostic bundle too early (such as during validation, before anything exists) or too late (after resources have been deleted).

Additionally, remove the 'rollback' flag of the CommandContext, and restructure so that destructive terminal tasks are not run in the create workflow. This allows us to run workflow steps (like the diagnostic one above) on error, which was previously only allowed if that flag was set to 'true'.

Additionally, remove an unused workflow and provider method for infrastructure cleanup.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/606#" title="Author self-approved">danbudris</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by abhay-krishna** [comment]
Skipping CI for Draft Pull Request.
If you want CI signal for your change, please convert it to an actual PR.
You can still manually trigger a test run with `/test all`

**Comment by abhay-krishna** [comment]
@danbudris: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
eks-anywhere-attribution-files-presubmit | e19f4696435d8616dc3f17f75d2beed3f7749490 | [link](https://prow.eks.amazonaws.com/view/s3/prowpresubmitsdataclusterstack-prowbucket7c73355c-vfwwxd2eb4gp/pr-logs/pull/aws_eks-anywhere/606/eks-anywhere-attribution-files-presubmit/1457845764012118016) | `/test eks-anywhere-attribution-files-presubmit`

[Full PR test history](https://prow.eks.amazonaws.com/pr-history?org=aws&repo=eks-anywhere&pr=606). [Your PR dashboard](https://prow.eks.amazonaws.com/pr?query=is:pr+state:open+author:danbudris).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Review by mdsgabriel — APPROVED** [review]

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/606#" title="Author self-approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/606#pullrequestreview-801736651" title="Approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/workflows/create.go
Comment (g-gaston) [inline pkg/workflows/create.go:141]: What about having a `CollectDiagnosticsTask` that has a next task field?
you could do `return &CollectDiagnosticsTask{nextTask: &DeleteKindClusterTask{})`

Comment (danbudris) [reply pkg/workflows/create.go:141]: ah yeah we could do that as well, I don't feel strongly either way

**Review by g-gaston — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/606#" title="Author self-approved">danbudris</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/606#pullrequestreview-802956181" title="LGTM">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/606#pullrequestreview-801736651" title="Approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [danbudris,g-gaston,mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
